spec/app/ics-027-interchain-accounts/README.md

```diff
@@ -185,28 +185,85 @@ An example of an active channel on the controller chain can look like this:
+ICS-04 allows for each channel version negotiation to be application-specific. In the case of interchain accounts, the channel version will be a string of a JSON struct containing all the relevant metadata intended to be relayed to the counterparty during the channel handshake step ([see summary below](#Metadata-negotiation-summary)).
 
-Due to how the mechanics of ICS-04 channel version negotiation operate the version passed into the host chain side (`OnChanOpenTry`) must match the counterparty version passed into the controller side (`OnChanOpenAck`) otherwise, there will be a resulting error at the IBC protocol level. 
+Combined with the one channel per interchain account approach, this method of metadata negotiation allows us to pass the address of the interchain account back to the controller chain and create a mapping from controller portID -> interchain account address during the `OnChanOpenAck` callback. As outlined in the [controlling flow](#Controlling-flow), a controller chain will need to know the address of a registered interchain account in order to send transactions to the account on the host chain.
 
-Combined with the one channel per interchain account approach, this method of version negotiation allows us to pass the address of the interchain account back to the controller chain and create a mapping from controller portID -> interchain account address during the `OnChanOpenAck` callback. As outlined in the [controlling flow](#Controlling-flow), a controller chain will need to know the address of a registered interchain account in order to send transactions to the account on the host chain.
-
-#### Version negotiation summary
+#### **Metadata negotiation summary**
 
 `interchain-account-address` is the address of the interchain account registered on the host chain by the controller chain.
 
-| Initiator | Datagram         | Chain acted upon |  Version (Controller, Host) |
-| --------- | ---------------- | ---------------- |  ---------------------- | 
-| Controller| ChanOpenInit     | Controller       |  (ics27-1, none)        | 
-| Relayer   | ChanOpenTry      | Host             |  (ics27-1, ics27-1.{interchain-account-address})        | 
-| Relayer   | ChanOpenAck      | Controller       |  (ics27-1, ics27-1.{interchain-account-address})        | 
-| Relayer   | ChanOpenConfirm  | Host             |  (ics27-1, ics27-1.{interchain-account-address})        | 
+**INIT**
+
+Initiator: Controller
+
+Datagram: ChanOpenInit
+
+Chain Acted Upon: Controller
+
+Version: 
+```json
+{
+  "Version": "ics27-1",
+  "Encoding": ["supported encoding types", "..."],
+  "TxTypes": ["supported transaction types", "..."],
```

Comment:
personally I'd prefer to keep this simple. INIT should propose an encoding/tx type, if the counterparty can't support them, it should abort. It won't always be clear what encoding type is best for this channel for the counterparty to select. The initiating actor should have the best insight, but in this situation also has the least control (outside of only setting one supported type - which is what I'm proposing)

Then we don't need `len(encoding) == 1` checks in the rest of the handshake

Is there a concrete example of when multiple supported encoding types may be useful to propose on INIT?

Reply:
The spec is as general as possible but I agree that the SDK implementation should be extremely simple. We should just include a single string not even in an array on the SDK.

Reply:
Both chains need to encode/decode the same way. Wouldn't one chain not using an array cause an issue?

Reply:
> The initiating actor should have the best insight, but in this situation also has the least control

Yea I suppose we can just limit to 1 type to ensure initiator either gets their preferred type or noting at all.